flox/xarray.py

```diff
@@ -19,7 +19,10 @@
 from .xrutils import _contains_cftime_datetimes, _to_pytimedelta, datetime_to_numeric
 
 if TYPE_CHECKING:
-    from xarray import DataArray, Dataset, Resample
+    from xarray import DataArray, Dataset  # TODO: Use T_DataArray, T_Dataset?
```

Comment:
Sure, we should explicitly say that `xarray.types` (is that right?) is public somewhere on the xarray docs.

Reply:
It's `xarray.core.types` so I suppose it's technically private at the moment. Maybe for the better? I don't think `.types` has settled enough yet to start recommending to the larger audience. Doesn't stop us from using it early though! :)

I mainly wrote the ToDo because I had issues with mypy, but this was the solution:
```python
# This errors if obj: T_Dataset | T_DataArray.
    if isinstance(obj, xr.DataArray):
        ds = obj._to_temp_dataset()
    else:
        ds = obj

# This passes if obj: T_Dataset | T_DataArray.
    if isinstance(obj, xr.Dataset):
        ds = obj
    else:
        ds = obj._to_temp_dataset()
```

Reply:
The other reason this is fine is that I'd like to move the contents of this file over to xarray in the long term.